crates/bevy_ui/src/focus.rs

```diff
@@ -216,20 +216,26 @@ pub fn ui_focus_system(
                     }
                 }
 
-                let position = node.global_transform.translation();
-                let ui_position = position.truncate();
-                let extents = node.node.size() / 2.0;
-                let mut min = ui_position - extents;
-                if let Some(clip) = node.calculated_clip {
-                    min = Vec2::max(min, clip.clip.min);
-                }
+                let node_rect = node.node.logical_rect(node.global_transform);
+
+                // If there is no `calculated_clip`, intersect with an unbounded `Rect`.
+                let clip_rect = node.calculated_clip.map(|clip| clip.clip).unwrap_or(Rect {
+                    min: Vec2::splat(f32::NEG_INFINITY),
+                    max: Vec2::splat(f32::INFINITY),
+                });
+
+                // Intersect with `clip_rect` to find the bounds of the visible region of the node
+                let visible_rect = node_rect.intersect(clip_rect);
```

Comment:
Would it make sense to create `visible_rect` from an if-let on the `calculated_clip` rather than create and intersect a dummy rect? I’m guessing the performance cost, if any, is negligible with what you have but just wanted to check.

Reply:
oh even better, there's no need to assign to clip rect at all, can intersect with node_rect in the calculated clip map closure and unwrap_or returns node_rect.